common/djangoapps/third_party_auth/tests/test_models.py

```diff
@@ -53,3 +58,91 @@ def test_clean_username_unicode_enabled(self):
         Test the username cleaner function with unicode enabled
         """
         assert clean_username('ItJüstWòrks™') == 'ItJüstWòrks'
+
+    def test_saml_configuration_dynamic_lookup(self):
+        """
+        Test SAML configuration lookup functionality, including the toggle behavior and error handling.
+        """
+        site = SiteFactory()
+
+        # Create initial SAML configuration
+        initial_config = SAMLConfiguration.objects.create(
+            site=site,
+            slug='test-config',
+            enabled=True,
+            entity_id='https://initial-entity-id.example.com',
+            private_key='initial_private_key',
+            public_key='initial_public_key',
+        )
+        initial_config_id = initial_config.id
+
+        # Create provider that points to the initial config
+        provider_config = SAMLProviderConfigFactory(
+            site=site,
+            entity_id='https://test-idp.example.com',
+            metadata_source='https://test-idp.example.com/metadata.xml',
+            saml_configuration=initial_config
+        )
+
+        # Create provider data to avoid AuthNotConfigured error
+        SAMLProviderData.objects.create(
+            entity_id=provider_config.entity_id,
+            sso_url='https://test-idp.example.com/SSO',
+            public_key='test_public_key',
+            fetched_at=timezone.now()
+        )
+
+        # Update configuration creates a new record with the same slug
+        initial_config.entity_id = 'https://updated-entity-id.example.com'
+        initial_config.change_date = timezone.now() + timedelta(minutes=5)
+        initial_config.save()
+        initial_config.refresh_from_db()
+
+        # Verify we have two configs with the same slug
```

Comment:
You can also look for the implementation of subTest to make the different cases more segregated but this is an optimization. If it takes more time, you can skip this.